### ISSUE TITLE
[master] Combined files are broken

### ISSUE THREAD

**Issue by Toflar** [contao/core-bundle#1475]
If you enable combine files in the page layout settings and run in prod environment, there are some funny urls created like so:

```
http://contao4.test/assets/css/fonts.css+mooRainbow.min.css+chosen.min.css+simplemodal.min.css
```

**Comment by fritzmg** [comment]
That's intentional ;). Duplicate of https://github.com/contao/core-bundle/issues/1423

**Comment by Toflar** [comment]
Yeah the url's don't bother me, it's not working for me ;-)
I found out why this is:

The combiner uses `+` as a separator which has a special meaning in an URL. It stands for a ` `. So if you don't access the server directly but pipe your assets through e.g. php that does urldecoding by default, it doesn't work.

If I do the following in the `Combiner.php`, it works just fine:

```diff
-$strKey = \StringUtil::substr(implode('+', $arrPrefix), 64, '...') . '-' . substr(md5($this->strKey), 0, 8);
+$strKey = \StringUtil::substr(implode('-', $arrPrefix), 64, '...') . '-' . substr(md5($this->strKey), 0, 8);
```

I think for the sake of compatibility, we should not use `+`. Wdyt?

**Comment by fritzmg** [comment]
I see, I haven't thought of that.

**Comment by Toflar** [comment]
Yeah I mean, I'm fine with the `+` but then the assets have to be stored using a space char on the server side :-)

**Comment by fritzmg** [comment]
Hm, but why is this not a problem in general? I mean it obviously works - at least with Apache. Is it web server specific?

**Comment by m-vo** [comment]
As Google also seems to use `,,` as a separator in pagespeed (and this was the reasoning in the other ticket), we could as well swap `+` by `,,`. :smile: 

But to be honest, I did not understand why long filenames would be better. And if so, what their sytax should look like to be actually helpful.

**Comment by leofeyer** [comment]
Ok, so we do have to replace `+`. It is not a problem for Google Pagespeed, because the URLs are not actually pointing to a file but are mapped to the Pagespeed cache by Nginx.

> But to be honest, I did not understand why long filenames would be better.

One of the reasons: Which of the following files contains the combined CSS framework files and which one contains the combined bundle style sheets?

```html
<link rel="stylesheet" href="assets/css/c6c50517.css">
<link rel="stylesheet" href="assets/css/9ae9863a.css">
```

You cannot tell. But now you can:

```html
<link rel="stylesheet" href="assets/css/fonts.css+basic.css+login.css-c6c50517.css">
<link rel="stylesheet" href="assets/css/faq.min.css+news.min.css+calendar.min.css-9ae9863a.css">
```

It is a bit like using Encore, which converts `app.css` into `app.65a84097165c7d093dcd5cc099aba1c6.css` so you still know which file it is.
